.github/workflows/release.yml

```diff
@@ -38,43 +36,56 @@ jobs:
 
       - name: Sign APK
         id: sign_apk
-        uses: r0adkll/sign-android-release@v1
-        with:
-          releaseDirectory: app/build/outputs/apk/release
-          signingKeyBase64: ${{ secrets.BASE64KEY }}
-          alias: ${{ secrets.ALIAS }}
-          keyStorePassword: ${{ secrets.KEY_STORE_PASSWORD }}
-          keyPassword: ${{ secrets.KEY_PASSWORD }}
+        shell: bash
+        run: |
+          APK_FILE=app/build/outputs/apk/release/app-release-unsigned.apk
+          SIGNED_APK_FILE=app/build/outputs/apk/release/app-release-signed.apk
+          echo ${{ secrets.BASE64KEY }} | base64 --decode > /tmp/signingkey.jks
+          apksigner=${ANDROID_HOME}/build-tools/$(ls ${ANDROID_HOME}/build-tools/ | tail -1)/apksigner
+          echo "APK Signer Path: $apksigner"
+          $apksigner sign \
+            --ks /tmp/signingkey.jks \
+            --ks-key-alias ${{ secrets.ALIAS }} \
+            --ks-pass pass:${{ secrets.KEY_STORE_PASSWORD }} \
+            --key-pass pass:${{ secrets.KEY_PASSWORD }} \
+            --out $SIGNED_APK_FILE \
+            --alignment-preserved true \
+            $APK_FILE
         env:
-          BUILD_TOOLS_VERSION: ${{ env.BUILD_TOOL_VERSION }}
-
+          ANDROID_HOME: ${{ env.ANDROID_HOME }}
+      
       - name: Make artifact
         uses: actions/upload-artifact@v2
         with:
           name: Signed APP APK
-          path: ${{steps.sign_apk.outputs.signedReleaseFile}}
+          path: app/build/outputs/apk/release/app-release-signed.apk
 
       - name: Build AAB
         run: bash ./gradlew bundleRelease
 
       - name: Sign AAB
         id: sign_aab
-        uses: r0adkll/sign-android-release@v1
-        with:
-          releaseDirectory: app/build/outputs/bundle/release
-          signingKeyBase64: ${{ secrets.BASE64KEY }}
-          alias: ${{ secrets.ALIAS }}
-          keyStorePassword: ${{ secrets.KEY_STORE_PASSWORD }}
-          keyPassword: ${{ secrets.KEY_PASSWORD }}
+        shell: bash
+        run: |
+          AAB_FILE=app/build/outputs/bundle/release/app-release.aab
+          SIGNED_AAB_FILE=app/build/outputs/bundle/release/app-release-signed.aab
+          echo ${{ secrets.BASE64KEY }} | base64 --decode > /tmp/signingkey.jks
```

Comment:
I think `secrets.BASE64KEY` likely contains newlines, and thus the script being run is:

```sh
APK_FILE=app/build/outputs/apk/release/app-release-unsigned.apk
SIGNED_APK_FILE=app/build/outputs/apk/release/app-release-signed.apk
echo LINE_1
LINE_2
LINE_3
...
LAST_LINE | base64 --decode > /tmp/signingkey.jks
...
```

And thus you get an error because the second line of the base64-encoded secret is not a valid command :)

Reply:
Shell scripting is tricky; environment variables and quoting are your friend :)

Reply:
Damn, thanks! That was indeed the issue. Now I finally got an error that makes sense. I'll try to fix it now.